zipline/data/_adjustments.pyx

```diff
@@ -234,64 +291,50 @@ cpdef load_adjustments_from_sqlite(object adjustments_db,  # sqlite3.Connection
-    for sid, ratio, eff_date in mergers:
-        if eff_date < start_date:
-            continue
+        if should_include_price_adjustments:
+            price_adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
+            price_adjustments.setdefault(date_loc, []).append(price_adj)
 
-        date_loc = _lookup_dt(date_ixs, eff_date, _dates_seconds)
+        if should_include_volume_adjustments:
+            volume_adj = Float64Multiply(
+                0, date_loc, asset_ix, asset_ix, 1.0 / ratio
+            )
+            volume_adjustments.setdefault(date_loc, []).append(volume_adj)
 
-        if not PyDict_Contains(asset_ixs, sid):
-            asset_ixs[sid] = assets.get_loc(sid)
-        asset_ix = asset_ixs[sid]
+    # mergers and dividends affect prices only
+    if should_include_price_adjustments:
+        for sid, ratio, eff_date in mergers:
+            if eff_date < start_date:
+                continue
 
-        adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
-        for i, column in enumerate(columns):
-            col_adjustments = results[i]
-            if column != 'volume':
-                try:
-                    col_adjustments[date_loc].append(adj)
-                except KeyError:
-                    col_adjustments[date_loc] = [adj]
-
-    # dividends affect prices only
-    for sid, ratio, eff_date in dividends:
-        if eff_date < start_date:
-            continue
+            date_loc = _lookup_dt(date_ixs, eff_date, _dates_seconds)
 
-        date_loc = _lookup_dt(date_ixs, eff_date, _dates_seconds)
+            if not PyDict_Contains(asset_ixs, sid):
+                asset_ixs[sid] = assets.get_loc(sid)
+            asset_ix = asset_ixs[sid]
 
-        if not PyDict_Contains(asset_ixs, sid):
-            asset_ixs[sid] = assets.get_loc(sid)
-        asset_ix = asset_ixs[sid]
+            price_adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
+            price_adjustments.setdefault(date_loc, []).append(price_adj)
+
+        for sid, ratio, eff_date in dividends:
+            if eff_date < start_date:
+                continue
+
+            date_loc = _lookup_dt(date_ixs, eff_date, _dates_seconds)
+
+            if not PyDict_Contains(asset_ixs, sid):
+                asset_ixs[sid] = assets.get_loc(sid)
+            asset_ix = asset_ixs[sid]
+
+            price_adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
+            price_adjustments.setdefault(date_loc, []).append(price_adj)
 
-        adj = Float64Multiply(0, date_loc, asset_ix, asset_ix, ratio)
-        for i, column in enumerate(columns):
-            col_adjustments = results[i]
-            if column != 'volume':
-                try:
-                    col_adjustments[date_loc].append(adj)
-                except KeyError:
-                    col_adjustments[date_loc] = [adj]
+    if should_include_price_adjustments:
+        result['price_adjustments'] = price_adjustments
```

Comment:
should these names match the input arguments like `P` or `V`, or should we use these names as the inputs? Also, we could probably drop the `_adjustments` suffix because it is clear from context.

Reply:
I think using `price` and `volume` would be better, I'll make that change.